app/fileManager.js

```diff
@@ -59,17 +69,62 @@ export default function(state, emitter) {
   function render() {
     emitter.emit('render');
   }
+  /*
+
+  function updateDloads() {
+    emitter.emit('updateDloads');
+  }
+*/
+
+  const dlimitArray = [];
+  const fileIdArray = [];
+  function ifRerender(file) {
+    if (fileIdArray.indexOf(file.id) === -1) {
+      const fileDetails = {
+        fileId: file.id,
+        dlimit: file.dlimit,
+        dtotal: file.dtotal
+      };
+      fileIdArray.push(file.id);
+      dlimitArray.push(fileDetails);
+      return false;
+    } else {
+      for (const fileDetail of dlimitArray) {
+        if (file.id === fileDetail.fileId) {
+          if (
+            file.dlimit !== fileDetail.dlimit ||
+            file.dtotal !== fileDetail.dtotal
+          ) {
+            fileDetail.dtotal = file.dtotal;
+            fileDetail.dlimit = file.dlimit;
+            return true;
+          } else {
+            return false;
+          }
+        }
+      }
+    }
+  }
 
   async function checkFiles() {
     const files = state.storage.files;
     let rerender = false;
+
     for (const file of files) {
-      const ok = await exists(file.id);
-      if (!ok) {
+      const receivedFile = await getDLCounts(file).then(function(data) {
+        return data;
+      });
+      console.log(receivedFile);
+      if (receivedFile.dlimit === receivedFile.dtotal) {
```

Comment:
I don't think we'll ever get here. The server deletes the file when dtotal = dlimit so we're likely to get a 404 instead.

Reply:
> if (receivedFile.dlimit === receivedFile.dtotal) {

Then what condition should be used here ?

Reply:
It depends on how you decide to implement `getDLCounts`. Hint: `FileReceiver.getMetadata` throws an `Error(404)` when the file isn't found.

Reply:
Oh thanx! Please have a look now